addons/binding/org.openhab.binding.blebox/src/main/java/org/openhab/binding/blebox/internal/discovery/BleboxScanner.java

```diff
@@ -0,0 +1,239 @@
+import org.apache.http.concurrent.FutureCallback;
+import org.openhab.binding.blebox.devices.DeviceInfo;
+import org.openhab.binding.blebox.devices.StatusResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link BleboxScanner} is responsible for scan local network for Blebox devices
+ *
+ * @author Szymon Tokarski - Initial contribution
+ */
+public class BleboxScanner {
+    private Logger logger = LoggerFactory.getLogger(BleboxScanner.class);
+
+    static final int ENVISALINK_BRIDGE_PORT = 4025;
+    static final int CONNECTION_TIMEOUT = 10;
+    static final int TIMEOUT = 2500;
+    static final String ENVISALINK_DISCOVERY_RESPONSE = "505";
+
+    private BleboxDiscovery bleboxDiscovery = null;
+
+    public BleboxScanner(BleboxDiscovery bleboxDiscovery) {
+        this.bleboxDiscovery = bleboxDiscovery;
+    }
+
+    /**
+     * Method for Bridge Discovery.
+     */
+    public synchronized void discoverBridge() {
+        logger.debug("Starting Blebox Discovery.");
+
+        SubnetUtils subnetUtils = null;
+        SubnetInfo subnetInfo = null;
+        long lowIP = 0;
+        long highIP = 0;
+
+        try {
+            List<Inet4Address> inet4 = getInet4Addresses();
+            logger.error("discoverBridge(): ip addresses - {}", inet4.size());
+            logger.error("discoverBridge(): ip main ip - {}", inet4.get(0));
+
+            NetworkInterface networkInterfacee = NetworkInterface.getByInetAddress(inet4.get(0));
+            logger.error("discoverBridge(): networkInterface - {}", networkInterfacee.toString());
+            String hostAddress = inet4.get(0).getHostAddress();
+            subnetUtils = new SubnetUtils(hostAddress + "/" + "24");
+            subnetInfo = subnetUtils.getInfo();
+            lowIP = convertIPToNumber(subnetInfo.getLowAddress());
+            highIP = convertIPToNumber(subnetInfo.getHighAddress());
+        } catch (IllegalArgumentException e) {
+            logger.error("discoverBridge(): Illegal Argument Exception - {}", e.toString());
+            return;
+        } catch (Exception e) {
+            logger.error("discoverBridge(): Error - Unable to get Subnet Information! {}", e.toString());
+            return;
+        }
+
+        logger.debug("   Local IP Address: {} - {}", subnetInfo.getAddress(),
+                convertIPToNumber(subnetInfo.getAddress()));
+        logger.debug("   Subnet:           {} - {}", subnetInfo.getNetworkAddress(),
+                convertIPToNumber(subnetInfo.getNetworkAddress()));
+        logger.debug("   Network Prefix:   {}", subnetInfo.getCidrSignature().split("/")[1]);
+        logger.debug("   Network Mask:     {}", subnetInfo.getNetmask());
+        logger.debug("   Low IP:           {}", convertNumberToIP(lowIP));
+        logger.debug("   High IP:          {}", convertNumberToIP(highIP));
+
+        ExecutorService threadpool = Executors.newFixedThreadPool(50);
```

Comment:
You're not allowed to create your own thread pool. Use the scheduler in the discovery service class.

Reply:
Network scanning with scheduler (single scan run with scheduler.execute()) took about 3 minutes to complete. Can I leave scanning with my own thread pool?

Reply:
Actually I'm not sure what the right implementation is. The coding guidelines are not completely clear on this, other than it mentions not to use your own threads or otherwise discuss it with the community. Because of this, and because I also saw similar implementations I asked about this on the forum. Theoretically you could schedule multiple calls too, but I'm no sure if that is the best solution. So for now I would suggest to just leave it as is, until maybe a better solution is suggested.

Reply:
Ok, thank you. So I'm leaving it with thread pool(50), network scanning tooks about 10 seconds in range 1-254